Pinta.Effects/Effects/MandelbrotFractalEffect.cs

```diff
@@ -53,77 +53,103 @@ private static double Mandelbrot (double r, double i, int factor)
 		int c = 0;
 		double x = 0;
 		double y = 0;
-
 		while ((c * factor) < 1024 && ((x * x) + (y * y)) < Max) {
 			double t = x;
-
-			x = x * x - y * y + r;
-			y = 2 * t * y + i;
-
+			x = (x * x) - (y * y) + r;
+			y = (2 * t * y) + i;
 			++c;
 		}
-
-		return c - Math.Log (y * y + x * x) * inv_log_max;
+		return c - Math.Log ((y * y) + (x * x)) * inv_log_max;
 	}
 
-	public override void Render (ImageSurface src, ImageSurface dst, ReadOnlySpan<RectangleI> rois)
+	private sealed record MandelbrotSettings (
+		int w,
+		int h,
+		double invH,
+		double invZoom,
+		double invQuality,
+		int count,
+		double invCount,
+		double angleTheta,
+		int factor,
+		double xOffset,
+		double yOffset);
+	private MandelbrotSettings CreateSettings (ImageSurface dst)
 	{
-		int w = dst.Width;
 		int h = dst.Height;
-
-		double invH = 1.0 / h;
 		double zoom = 1 + zoom_factor * Data.Zoom;
-		double invZoom = 1.0 / zoom;
-
-		double invQuality = 1.0 / (double) Data.Quality;
-
 		int count = Data.Quality * Data.Quality + 1;
-		double invCount = 1.0 / (double) count;
-		double angleTheta = (Data.Angle * 2 * Math.PI) / 360;
 
-		Span<ColorBgra> dst_data = dst.GetPixelData ();
-		int dst_width = dst.Width;
+		return new (
 
-		foreach (Core.RectangleI rect in rois) {
-			for (int y = rect.Top; y <= rect.Bottom; y++) {
-				var dst_row = dst_data.Slice (y * dst_width, dst_width);
+			w: dst.Width,
+			h: h,
 
-				for (int x = rect.Left; x <= rect.Right; x++) {
-					int r = 0;
-					int g = 0;
-					int b = 0;
-					int a = 0;
+			invH: 1.0 / h,
+			invZoom: 1.0 / zoom,
 
-					for (double i = 0; i < count; i++) {
-						double u = (2.0 * x - w + (i * invCount)) * invH;
-						double v = (2.0 * y - h + ((i * invQuality) % 1)) * invH;
+			invQuality: 1.0 / (double) Data.Quality,
 
-						double radius = Math.Sqrt ((u * u) + (v * v));
-						double radiusP = radius;
-						double theta = Math.Atan2 (v, u);
-						double thetaP = theta + angleTheta;
+			count: count,
+			invCount: 1.0 / (double) count,
+			angleTheta: (Data.Angle * 2 * Math.PI) / 360,
 
-						double uP = radiusP * Math.Cos (thetaP);
-						double vP = radiusP * Math.Sin (thetaP);
+			factor: Data.Factor,
 
-						double m = Mandelbrot ((uP * invZoom) + this.x_offset, (vP * invZoom) + this.y_offset, Data.Factor);
+			xOffset: x_offset,
+			yOffset: y_offset
+		);
+	}
 
-						double c = 64 + Data.Factor * m;
+	private readonly record struct LoopContext (int y, int x);
```

Comment:
Similar to the other pull request, I think `GetPixelColor` can either take separate x / y parameters or a `PointI`